src/ucp/rndv/rndv_get.c

```diff
@@ -71,13 +71,19 @@ ucp_proto_rndv_get_common_probe(const ucp_proto_init_params_t *init_params,
     }
 
     status = ucp_proto_rndv_bulk_init(&params, UCP_PROTO_RNDV_GET_DESC,
-                                      UCP_PROTO_RNDV_ATS_NAME, &rpriv, &caps);
+                                      UCP_PROTO_RNDV_ATS_NAME, &perf, &rpriv);
     if (status != UCS_OK) {
         return;
     }
 
     priv_size = UCP_PROTO_MULTI_EXTENDED_PRIV_SIZE(&rpriv, mpriv);
-    ucp_proto_common_add_proto(&params.super, &caps, &rpriv, priv_size);
+    status    = ucp_proto_select_add_proto(&params.super.super,
+                                           params.super.cfg_thresh,
+                                           params.super.cfg_priority, perf,
+                                           &rpriv, priv_size);
+    if (status != UCS_OK) {
+        ucp_proto_perf_destroy(perf);
+    }
```

Comment:
maybe should add a wrapper to destroy perf if failed to add?
or maybe ucp_proto_select_add_proto should return void and destroy perf itself if failed to add?
since this construct repeats itself in multiple places

Reply:
I had such wrapper before but @rakhmets mentioned that this makes things less straightforward and adds more error 
prone since we create perf on one level and destroy it on another. See here https://github.com/openucx/ucx/pull/10051#discussion_r1722944168

Reply:
Changed back to destroying perf inside `ucp_proto_select_add_proto`